x-pack/plugins/viz_editor/public/common/lib/vis_model.ts

```diff
@@ -4,48 +4,38 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { SelectOperation } from '../../common';
+import { Query } from '../../../common';
 
-export interface IndexPatternField {
+export interface Field {
   name: string;
   type: string;
   aggregatable: boolean;
   searchable: boolean;
 }
 
-export interface IndexPattern {
+export interface Datasource<M = any> {
   id: string;
   title: string;
   timeFieldName?: string;
-  fields: IndexPatternField[];
+  fields: Field[];
   fieldFormatMap?: string;
-}
-
-export interface VisModelQuery {
-  select: {
-    [id: string]: SelectOperation;
-  };
+  meta?: M;
```

Comment:
I'm not sure every data source is going to conform to this interface, but I'm not you need to make any change in this PR.

Reply:
The idea was that everything looks like an index-pattern to the rest of the editor - and specific stuff goes into meta (e.g. the sql for the essql datasource)